code/game/objects/items/cash.dm

```diff
@@ -0,0 +1,193 @@
+	resistance_flags = FLAMMABLE
+	var/value = 0
+	grind_results = list(/datum/reagent/iron = 10)
+
+/obj/item/spacecash/Initialize(mapload, amount)
+	..()
+	if(amount)
+		value = amount
+	update_icon()
+
+/obj/item/spacecash/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/spacecash))
+		var/obj/item/spacecash/bundle/bundle
+		if(istype(W, /obj/item/spacecash/bundle))
+			bundle = W
+		else
+			var/obj/item/spacecash/cash = W
+			bundle = new (loc)
+			bundle.value = cash.value
+			user.dropItemToGround(cash)
+			qdel(cash)
+
+		bundle.value += value
+		bundle.update_icon()
+		if(ishuman(user))
+			var/mob/living/carbon/human/H = user
+			H.dropItemToGround(src)
+			H.dropItemToGround(bundle)
+			H.put_in_hands(bundle)
+		to_chat(user, "<span class='notice'>You add [value] credits worth of money to the bundle.<br>It now holds [bundle.value] credits.</span>")
+		qdel(src)
+
+/obj/item/spacecash/Destroy()
+	. = ..()
+	value = 0 // Prevents money from be duplicated anytime.//I'll trust eris on this one
+
+/obj/item/spacecash/bundle
+	icon_state = "nothing"
+
+/obj/item/spacecash/bundle/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/item/spacecash/bundle/update_icon()
+	cut_overlays()
+	var/remaining_value = value
+	var/iteration = 0
+	var/coins_only = TRUE
+	var/list/coin_denominations = list(10, 5, 1)
+	var/list/banknote_denominations = list(1000, 500, 200, 100, 50, 20)
+	for(var/i in banknote_denominations)
+		while(remaining_value >= i && iteration < 50)
+			remaining_value -= i
+			iteration++
+			var/image/banknote = image('icons/obj/economy.dmi', "credit[i]")
+			var/matrix/M = matrix()
+			M.Translate(rand(-6, 6), rand(-4, 8))
+			banknote.transform = M
+			overlays += banknote
+			coins_only = FALSE
+
+	if(remaining_value)
+		for(var/i in coin_denominations)
+			while(remaining_value >= i && iteration < 50)
+				remaining_value -= i
+				iteration++
+				var/image/coin = image('icons/obj/economy.dmi', "credit[i]")
+				var/matrix/M = matrix()
+				M.Translate(rand(-6, 6), rand(-4, 8))
+				coin.transform = M
+				overlays += coin
+
+	if(coins_only)
+		if(value == 1)
+			name = "one credit coin"
+			desc = "Heavier then it looks."
+			drop_sound = 'sound/items/handling/coin_drop.ogg'
+			pickup_sound =  'sound/items/handling/coin_pickup.ogg'
+		else
+			name = "[value] credits"
+			desc = "Heavier than they look."
+			gender = PLURAL
+			drop_sound = 'sound/items/handling/coin_drop.ogg'
+			pickup_sound =  'sound/items/handling/coin_pickup.ogg'
+	else
+		if(value <= 3000)
+			name = "[value] credits"
+			gender = NEUTER
+			desc = "Some cold, hard cash."
+			drop_sound = 'sound/items/handling/dosh_drop.ogg'
+			pickup_sound =  'sound/items/handling/dosh_pickup.ogg'
+		else
+			name = "[value] credits"
+			gender = NEUTER
+			desc = "An executive's ransom made manifest."
+			drop_sound = 'sound/items/handling/dosh_drop.ogg'
+			pickup_sound =  'sound/items/handling/dosh_pickup.ogg'
+
+/obj/item/spacecash/bundle/attack_self()
```

Comment:
Can we get an altclick to do this same behaviour? It'd certainly be nice to have to draw from briefcases, piles on the ground, etc.

Reply:
an almost utter copy-paste into /altclick() has achieved this functionality perfectly
pushing now